### PR TITLE
Crosslink reassign

### PR DESCRIPTION
**Pull request by lonelu**
Avoid dead-end are identified as crosslink:
1. alpha: DEVVKK(5)    score: 7.038241      
beta: LHDYYK(6)     score: 5.051740931

2. LHDYYK[Crosslink:H2O   Dead End on X]DEVVKK(6)  score: 10.10030922

This one is dead-end, but the crosslink happened to have a higher score. This situation is reported by Rebecca.

### CODE REVIEW

**Comment by lonelu** [comment]
This pull request will be changed when PEP for crosslink is ready in the future.

File: TaskLayer/XLSearchTask/XLSearchTask.cs [hunk 1]
```diff
@@ -145,7 +145,23 @@ protected override MyTaskResults RunSpecific(string OutputFolder, List<DbForTask
                 filteredAllPsms.Add(RemoveDuplicateFromPsmsPerScan(psmsPerScan));
             }
 
-            var allPsmsList = filteredAllPsms.Select(p => p.First()).OrderByDescending(p => p.XLTotalScore).ToList();
+            var allPsmsList = new List<CrosslinkSpectralMatch>();
```
Comment (rmillikin) [inline TaskLayer/XLSearchTask/XLSearchTask.cs]: can you add a comment explaining the `List<List<CrosslinkSpectralMatch>>`? I don't understand why there is a list of lists

File: TaskLayer/XLSearchTask/XLSearchTask.cs [hunk 2]
```diff
@@ -145,7 +145,23 @@ protected override MyTaskResults RunSpecific(string OutputFolder, List<DbForTask
...
+
+            foreach (var psmsPerScan in allPsms)
```
Comment (rmillikin) [inline TaskLayer/XLSearchTask/XLSearchTask.cs]: Add comments to explain, I don't understand what is happening

File: TaskLayer/XLSearchTask/XLSearchTask.cs (hunk 1 above)
Comment (lonelu) [reply TaskLayer/XLSearchTask/XLSearchTask.cs]: Comment added.

File: TaskLayer/XLSearchTask/XLSearchTask.cs (hunk 2 above)
Comment (lonelu) [reply TaskLayer/XLSearchTask/XLSearchTask.cs]: Comment added

**Pull request merged**